x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/Mapper.java

```diff
@@ -104,6 +106,46 @@ public PhysicalPlan map(LogicalPlan p) {
         //
         // Unary Plan
         //
+        if (localMode == false && p instanceof Enrich enrich && enrich.mode() == Enrich.Mode.REMOTE) {
+            // When we have remote enrich, we want to put it under FragmentExec, so it would be executed remotely.
+            // We're only going to do it on the coordinator node.
+            // The way we're going to do it is as follows:
+            // 1. Locate FragmentExec in the tree. If we have no FragmentExec, we won't do anything.
+            // 2. Put this Enrich under it, removing everything that was below it previously.
```

Comment:
> removing everything that was below it previously
This most certainly sounds like a no-go. Do you mean insert the node under FragmentExec and the node below maybe?

Reply:
Everything that is under the Enrich will still be under it when it's inserted into the Fragment, as a logical plan (which will later be converted to physical when processing the fragment as I understand).